packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -35,6 +39,20 @@ const reservedWords = new Set([
 
 const PRIMITIVE_TYPES = new Set(['string', 'number', 'bigint', 'boolean', 'undefined', 'symbol', 'null'])
 
+function templateRequiresEvaluation (segments) {
+  if (segments === undefined) return false // There should always be segments, but just in case
+  for (const { str } of segments) {
+    if (str === undefined) return true
+  }
+  return false
+}
+
+function compileSegments (segments) {
+  return segments
+    .map(({ str, json }) => str !== undefined ? str : `\${${compile(json)}}`)
```

Comment:
we have different rules for serialization of object in templates. for example, objects here would be [object object]. but we would like to have `{ prop:value, probe:value, ...}` for the first level props.

Reply:
Not sure I understand? I assume `str` will always be a string and I assume `json` will always be the EL AST?

Reply:
Update: This was discussed on Slack and the issue is as follows: If someone writes a template where the evaluation of a segment doesn't returns an object instead for example a string or a number, that object with be stringified and the result in the log message will just be `[object Object]` (which is what you get if you evaluate `${{}}` inside of a template string).

We have decided not to deal with that issue in this PR, but make a follow up PR that properly stringifies these objects. Most likely using `util.inspect`.

Reply:
Update: I've updated the PR with support for this anyway. It's all pushed now 👍